File: scripts/check_integrity.ts
async function checkIntegrity() {
    const htmlPath = 'src/it-slider.html';
    const tsPath = 'src/app.ts';
    
    try {
        if (!(await exists(htmlPath)) || !(await exists(tsPath))) {
            console.error("Files not found.");
            Deno.exit(1);
        }

        const htmlContent = await Deno.readTextFile(htmlPath);
        const tsContent = await Deno.readTextFile(tsPath);

        // Find all IDs in HTML
        const htmlIdMatches = htmlContent.matchAll(/id=["']([^"']+)["']/g);
        const htmlIds = new Set([...htmlIdMatches].map(m => m[1]));
        
        // Find all getElementById calls in TS
        const tsIdMatches = tsContent.matchAll(/getElementById\(['"]([^'"]+)['"]\)/g);
        const tsIds = new Set([...tsIdMatches].map(m => m[1]));
        
        const missingInHtml = [...tsIds].filter(id => !htmlIds.has(id));
        
        // Filter out dynamic IDs like slider0, val0, cfgStart0, cfgMax0
        const filteredMissing = missingInHtml.filter(id => {
            const prefixes = ['slider', 'val', 'cfgStart', 'cfgMax'];
            return !prefixes.some(p => id.startsWith(p) && /^\d+$/.test(id.slice(p.length)));
        });

        if (filteredMissing.length === 0) {
            console.log("Integrity check passed (static IDs).");
        } else {
            console.error(`Missing IDs in HTML: ${filteredMissing}`);
            Deno.exit(1);
        }
    } catch (err) {
        console.error(`Error: ${err.message}`);
        Deno.exit(1);
    }
}

async function exists(filename: string): Promise<boolean> {
    try {
        await Deno.stat(filename);
        return true;
    } catch {
        return false;
    }
}

if (import.meta.main) {
    checkIntegrity();
}

File: check_integrity.py
import re
import os

def check_integrity():
    html_path = 'it-slider.html'
    ts_path = 'app.ts'
    
    if not os.path.exists(html_path) or not os.path.exists(ts_path):
        print("Files not found.")
        return

    with open(html_path, 'r', encoding='utf-8') as f:
        html_content = f.read()

    with open(ts_path, 'r', encoding='utf-8') as f:
        ts_content = f.read()

    # Find all IDs in HTML
    html_ids = set(re.findall(r'id=["\']([^"\']+)["\']', html_content))
    
    # Find all getElementById calls in TS
    ts_ids = set(re.findall(r"getElementById\(['\"]([^'\"]+)['\"]\)", ts_content))
    
    missing_in_html = ts_ids - html_ids
    
    # Filter out dynamic IDs like slider0, val0, cfgStart0, cfgMax0
    filtered_missing = [id for id in missing_in_html if not any(id.startswith(p) and id[len(p):].isdigit() for p in ['slider', 'val', 'cfgStart', 'cfgMax'])]

    if not filtered_missing:
        print("Integrity check passed (static IDs).")
    else:
        print(f"Missing IDs in HTML: {filtered_missing}")
        exit(1)

if __name__ == "__main__":
    check_integrity()
